server/routes/choropleth.py

```diff
@@ -0,0 +1,215 @@
+                        "geoId": geo_id,
+                    }
+                }
+            # Load, simplify, and add geoJSON coordinates.
+            # First returned value is chosen always.
+            # TODO(iancostello): Implement a better approach.
+            geojson = json.loads(json_text[0])
+            geo_feature['geometry']['coordinates'] = (
+                coerce_geojson_to_righthand_rule(
+                                    geojson['coordinates'],
+                                    geojson['type']))
+            # Process Statistical Observation if valid.
+            if ('data' in stat_var_by_geo.get(geo_id, []) and
+                    (not per_capita or
+                    'data' in population_by_geo.get(geo_id, []))):
+                # Grab the latest available data.
+                stat_obs = next(iter(
+                    reversed(stat_var_by_geo[geo_id]['data'].values())))
+
+                # Handle per-capita option.
+                if per_capita:
+                    # Note that this could be from a different period from the
+                    # statistical variable.
+                    # TODO(iancostello): Determine how to handle populations 
+                    # and statistical values from different times.
+                    pop = next(iter(
+                        reversed(population_by_geo[geo_id]['data'].values())))
+                    stat_obs /= pop
+
+                values.append(stat_obs)
+                geo_feature["properties"]["value"] = stat_obs
+
+            # Add to main dataframe.
+            features.append(geo_feature)
+    # Generate plotting information from value list.
+    domain, palette = determine_color_palette(values, per_capita)
+
+    # Return as json payload.
+    return jsonify({
+        "geoJson": {
+            "type": "FeatureCollection",
+            "features": features
+        },
+        "_PLOTTING_INFO": {
+            "domain": domain,
+            "palette": palette
+        }  
+    }, 200)
+
+def coerce_geojson_to_righthand_rule(geoJsonCords, obj_type):
+    """Changes GeoJSON handedness to the right-hand rule.
+    
+    GeoJSON is stored in DataCommons in the reverse format of what D3
+    expects. This results in geographies geometry being inverted. This function
+    fixes these lists to be in the format expected by D3 and turns all polygons
+    into multipolygons for downstream consistency.
+        Args:
+            geoJsonCords -> Nested list of geojson.
+            obj_type -> Object feature type.
+        Returns:
+            Nested list of geocoords.
+    """
+    if obj_type == "Polygon":
+        geoJsonCords[0].reverse()
+        return [geoJsonCords]
+    elif obj_type == "MultiPolygon":
+        for polygon in geoJsonCords:
+            polygon[0].reverse()
+        return geoJsonCords
+    else:
+        assert False, f"Type {obj_type} unknown!"
+
+def determine_color_palette(values, is_denominated):
```

Comment:
should the color rendering be moved to client side using d3?

Reply:
Will add a todo.